src/modules/Game.class.js

```diff
@@ -6,63 +6,261 @@
  * Feel free to add more props and methods if needed.
  */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
-  }
-
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  static gameStatuses = {
+    idle: 'idle',
+    playing: 'playing',
+    lose: 'lose',
+    win: 'win',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState;
+    this.state = this.initialState.map((row) => [...row]);
+    this.status = Game.gameStatuses.idle;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const updatedState = this.state.map((row) => this.move(row));
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(updatedState)) {
```

Comment:
Comparing the state before and after the move using 'JSON.stringify' is not the most efficient way. It's better to use a different approach to check for changes in the state, as 'JSON.stringify' can be expensive in terms of performance.